src/Microsoft.Azure.WebJobs.Host/Extensions/CloudQueueMessageExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.WindowsAzure.Storage.Queue
+{
+    internal static class CloudQueueMessageExtensions
+    {
+        private static PropertyHelper _idProp;
+        private static PropertyHelper _popReceiptProp;
+        private static PropertyHelper _insertionTimeProp;
+        private static PropertyHelper _nextVisibleTimeProp;
+        private static PropertyHelper _expirationTimeProp;
+
+        static CloudQueueMessageExtensions()
+        {
+            IReadOnlyList<PropertyHelper> messageProperties = PropertyHelper.GetProperties(typeof(CloudQueueMessage));
+            _idProp = messageProperties.Single(p => p.Name == nameof(CloudQueueMessage.Id));
+            _popReceiptProp = messageProperties.Single(p => p.Name == nameof(CloudQueueMessage.PopReceipt));
+            _insertionTimeProp = messageProperties.SingleOrDefault(p => p.Name == nameof(CloudQueueMessage.InsertionTime));
+            _nextVisibleTimeProp = messageProperties.SingleOrDefault(p => p.Name == nameof(CloudQueueMessage.NextVisibleTime));
+            _expirationTimeProp = messageProperties.SingleOrDefault(p => p.Name == nameof(CloudQueueMessage.ExpirationTime));
+        }
+
+        /// <summary>
+        /// Overwrites the message's properties with the specified values, if different.
+        /// </summary>
+        public static void SetProperties(this CloudQueueMessage message, string id, string popReceipt, DateTimeOffset? insertionTime,
```

Comment:
nit: now that this is conditional, I'd rename to UpdateChangedProperties or something